prov/rxm/src/rxm_eq.c

```diff
@@ -108,10 +115,11 @@ ssize_t rxm_eq_write(struct fid_eq *eq_fid, uint32_t event,
 		else {
 			rxm_mc->state = RXM_MC_OFF_STARTED;
 			ofi_spin_unlock(&rxm_mc->state_lock);
+			peer_context.mc_fid = &rxm_mc->mc_fid;
 			ret = fi_join_collective(rxm_mc->ep->offload_coll_ep,
 				rxm_mc->coll_addr, &rxm_mc->av_set->av_set_fid,
-				flags, &rxm_mc->offload_coll_mc_fid,
-				&rxm_mc);
+				flags | FI_PEER, &rxm_mc->offload_coll_mc_fid,
+				&peer_context);
```

Comment:
Similar - we can't pass a stack address to an async call.

Reply:
This has been modeled on rxm_cq_open().
The FI_PEER flag informs us to handle the context in a different way (not as persistent one)